src/main/webapp/app/entities/programming-exercise/test-cases/programming-exercise-manage-test-cases.component.ts

```diff
@@ -0,0 +1,126 @@
+    styleUrls: ['./programming-exercise-manage-test-cases.scss'],
+})
+export class ProgrammingExerciseManageTestCasesComponent implements OnInit, OnDestroy {
+    @ViewChild('editingInput', { static: false }) editingInput: ElementRef;
+    exerciseId: number;
+    editing: ProgrammingExerciseTestCase | null = null;
+    testCaseSubscription: Subscription;
+    paramSub: Subscription;
+
+    testCases: ProgrammingExerciseTestCase[] = [];
+    filteredTestCases: ProgrammingExerciseTestCase[] = [];
+
+    showInactiveValue = false;
+
+    get showInactive() {
+        return this.showInactiveValue;
+    }
+
+    set showInactive(showInactive: boolean) {
+        this.editing = null;
+        this.showInactiveValue = showInactive;
+        this.updateTestCaseFilter();
+    }
+
+    constructor(private testCaseService: ProgrammingExerciseTestCaseService, private route: ActivatedRoute) {}
+
+    /**
+     * Subscribes to the route params to get the current exerciseId.
+     * Uses the exerciseId to subscribe to the newest value of the exercise's test cases.
+     */
+    ngOnInit(): void {
+        this.paramSub = this.route.params.pipe(distinctUntilChanged()).subscribe(params => {
+            this.exerciseId = Number(params['exerciseId']);
+            this.editing = null;
+            if (this.testCaseSubscription) {
+                this.testCaseSubscription.unsubscribe();
+            }
+            this.testCaseSubscription = this.testCaseService.subscribeForTestCases(this.exerciseId).subscribe((testCases: ProgrammingExerciseTestCase[]) => {
+                this.testCases = testCases;
+                this.updateTestCaseFilter();
+            });
+        });
+    }
+
+    ngOnDestroy(): void {
+        if (this.testCaseSubscription) {
+            this.testCaseSubscription.unsubscribe();
+        }
+        if (this.paramSub) {
+            this.paramSub.unsubscribe();
+        }
+    }
+
+    /**
+     * Show an input to edit the test cases weight.
+     * @param rowIndex
+     */
+    enterEditing(rowIndex: number) {
+        this.editing = this.filteredTestCases[rowIndex];
+        setTimeout(() => {
+            if (this.editingInput) {
+                this.editingInput.nativeElement.focus();
+            }
+        });
+    }
+
+    /**
+     * Hide input.
+     */
+    leaveEditingWithoutSaving() {
+        this.editing = null;
+    }
+
+    /**
+     * Update the weight of the edited test case.
+     * @param event
+     */
+    updateWeight(event: any) {
+        if (!this.editing) {
+            return;
+        }
+        const editedTestCase = this.editing;
+        const weight = event.target.value;
+        this.testCaseService.updateWeight(this.exerciseId, editedTestCase.id, weight).subscribe((updatedTestCase: ProgrammingExerciseTestCase) => {
+            this.editing = null;
+            const updatedTestCases = this.testCases.map(testCase => (testCase.id === updatedTestCase.id ? updatedTestCase : testCase));
```

Comment:
What happens if this does not work? For example, because I am offline or the server has an error?
Even if I like the usability here, I think we should have a save button and update all test cases so that we can handle errors in a better way

Reply:
You are right, there was no error handling, I added it.

About updateWeight vs updateWeights: It would be better from a efficiency standpoint but not from a complexity standpoint.
See also the comment above. It's very easy to handle a single update, but when we decide to update all test cases at once, the implementation will become more complex. I'm not sure if this is the correct trade-off atm.

Reply:
See above comment.